plaid/src/apis/npm/npm_web_client.rs

```diff
@@ -0,0 +1,833 @@
+            .send()
+            .await
+            .map(|_| Ok(0))
+            .map_err(|_| ApiError::NpmError(NpmError::FailedToDeletePackage))?
+    }
+
+    /// Add a user to an npm team
+    pub async fn add_user_to_team(&self, params: &str, module: &str) -> Result<i32, ApiError> {
+        self.login()
+            .await
+            .map_err(|_| ApiError::NpmError(NpmError::LoginFlowError))?;
+        let params: AddRemoveUserToFromTeamParams =
+            serde_json::from_str(params).map_err(|_| ApiError::BadRequest)?;
+        let user = self.validate_npm_username(&params.user)?;
+        let team = self.validate_npm_team_name(&params.team)?;
+
+        info!(
+            "Adding user [{}] to team [{}] on behalf of [{module}]",
+            user, team
+        );
+
+        let csrf_token = self
+            .get_csrftoken_from_cookies()
+            .map_err(|_| ApiError::NpmError(NpmError::WrongClientStatus))?;
+        let body = format!(r#"{{"csrftoken": "{}", "user": "{}"}}"#, &csrf_token, user);
+        self.client
+            .post(format!(
+                "{}/settings/{}/teams/team/{}/users",
+                NPMJS_COM_URL, self.config.npm_scope, team
+            ))
+            .header("Content-Type", "text/plain;charset=UTF-8")
+            .body(body)
+            .send()
+            .await
+            .map(|_| Ok(0))
+            .map_err(|_| ApiError::NpmError(NpmError::FailedToAddUserToTeam))?
+    }
+
+    /// Remove a user from an npm team
+    pub async fn remove_user_from_team(&self, params: &str, module: &str) -> Result<i32, ApiError> {
+        self.login()
+            .await
+            .map_err(|_| ApiError::NpmError(NpmError::LoginFlowError))?;
+        let params: AddRemoveUserToFromTeamParams =
+            serde_json::from_str(params).map_err(|_| ApiError::BadRequest)?;
+        let user = self.validate_npm_username(&params.user)?;
+        let team = self.validate_npm_team_name(&params.team)?;
+
+        info!("Removing [{user}] from [{team}] on behalf of [{module}]");
+
+        let body = format!(
+            r#"{{"csrftoken": "{}"}}"#,
+            self.get_csrftoken_from_cookies()
+                .map_err(|_| ApiError::NpmError(NpmError::WrongClientStatus))?
+        );
+        self.client
+            .post(format!(
+                "{}/settings/{}/teams/team/{}/users/{}/delete",
+                NPMJS_COM_URL, self.config.npm_scope, team, user
+            ))
+            .header("Content-Type", "text/plain;charset=UTF-8")
+            .body(body)
+            .send()
+            .await
+            .map(|_| Ok(0))
+            .map_err(|_| ApiError::NpmError(NpmError::FailedToRemoveUserFromTeam))?
+    }
+
+    /// Remove a user from the npm organization
+    pub async fn remove_user_from_organization(
+        &self,
+        user: &str,
+        module: &str,
+    ) -> Result<i32, ApiError> {
+        #[derive(Serialize)]
+        struct RemoveUserPayload<'a> {
```

Comment:
Yes, I would pull this out so it can be used in other functions seeing as we have to do this a few times.

Reply:
Done. Turns out this struct is the same as the one we serialize for removing a user from a team. But I am reluctant to fuse them together because things might change in the future, and we don't want to couple these requests at all costs.

Reply:
That said, it's always about user removal from somewhere... I guess I will merge them 😄